go-apps/meep-tc-engine/server/tc-engine.go

```diff
@@ -197,27 +202,43 @@ var svcCountReq = 0
 var svcCount = 0
 var nextTransactionId = 1
 
+const redisAddr string = "meep-redis-master:6379"
+
+var rc *redis.Connector
+
+const DEFAULT_TC_ENGINE_DB = 0
+
 // Init - TC Engine initialization
 func Init() (err error) {
 
 	// Connect to Redis DB
-	err = DBConnect()
+	rc, err = redis.NewConnector(redisAddr, DEFAULT_TC_ENGINE_DB)
 	if err != nil {
-		log.Error("Failed connection to Active DB. Error: ", err)
+		log.Error("Failed connection to Redis DB.  Error: ", err)
 		return err
 	}
-	log.Info("Connected to Active DB")
+	log.Info("Connected to redis DB")
 
 	// Subscribe to Pub-Sub events for MEEP Controller
 	// NOTE: Current implementation is RedisDB Pub-Sub
-	err = Subscribe(channelCtrlActive, channelMgManagerLb)
+	err = rc.Subscribe(channelCtrlActive, channelMgManagerLb)
 	if err != nil {
 		log.Error("Failed to subscribe to Pub/Sub events. Error: ", err)
 		return err
 	}
 
 	// Flush any remaining TC Engine rules
-	DBFlush(moduleTcEngine)
+	rc.DBFlush(moduleTcEngine)
+	rc.DBFlush(moduleMetrics)
+
+	bwSharing, err = bws.NewBwSharing("default", updateOneFilterRule, applyOneFilterRule)
+
+	if err != nil {
+		log.Error("Failed to create a bwSharing object. Error: ", err)
+		return err
+	}
+	bwSharing.UpdateControls()
+	_ = bwSharing.Start()
 
```

Comment:
I recommend not starting BW Sharing until the following changes are made:
>NOTE: PR can be merged with disabled code for now. The following updates can be addressed in a subsequent PR.

1. **TC Engine must hand off Throughput management to BW Sharing object when it is started**
   1. TC Engine algorithm & BW Sharing algorithm both calculate the desired throughput to set
   1. When BW Sharing is enabled, the TC Engine should let the BW Sharing algorithm take control
1. **BW Sharing callback functions in TC Engine should be thread safe**
   1. TC Engine runs in active scenario listener thread context
   1. BW Sharing callback functions run in periodic ticker thread context
   1. TC Engine & callback functions both change the TC Engine filter & shaping rules in Redis DB
   1. Mutex should be added to protect data & DB access to avoid concurrency issues
1. **BW Sharing instance should listen for active scenario updates**
   1. Currently active scenario is passed from TC engine to BW Sharing object where it gets parsed to update its segments & network characteristics
   1. To be more in line with model refactoring implementation, we should remove the model from the BW Sharing API and listen for updates directly from the BW Sharing package instead
1. **BW Sharing algorithm should be implemented as a class**
   1. Listener thread and Periodic Ticker thread should invoke an object method instead of a static function
   1. Algorithm variables should be inside an Algorithm object with a stable interface

Reply:
Object creation left intact, but bws starting code commented out

Reply:
I added the above suggestions to Tier 2 implementation feature status.